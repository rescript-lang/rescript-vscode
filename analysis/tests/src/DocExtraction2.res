type t = string

let getStr = () => "123"

let make = getStr

module InnerModule = {
  type t = unit
  let make = () => ()
}

// ^dex
