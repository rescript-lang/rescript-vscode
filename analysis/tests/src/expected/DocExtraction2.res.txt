Documentation extraction src/DocExtraction2.res
extracting docs for src/DocExtraction2.res
preferring found resi file for impl: src/DocExtraction2.resi

{
  "name": "DocExtraction2",
  "docstrings": ["Module level doc here."],
  "items": [
  {
    "id": "DocExtraction2.t",
    "kind": "type",
    "name": "t",
    "signature": "type t",
    "docstrings": ["Type t is pretty cool."]
  }, 
  {
    "id": "DocExtraction2.make",
    "kind": "value",
    "name": "make",
    "signature": "let make: unit => t",
    "docstrings": ["Makerz of stuffz."]
  }, 
  {
    "id": "DocExtraction2.InnerModule",
    "name": "InnerModule",
    "kind": "module",
    "docstrings": [],
    "items": [
    {
      "id": "DocExtraction2.InnerModule.t",
      "kind": "type",
      "name": "t",
      "signature": "type t",
      "docstrings": ["This type is also t."]
    }, 
    {
      "id": "DocExtraction2.InnerModule.make",
      "kind": "value",
      "name": "make",
      "signature": "let make: unit => t",
      "docstrings": ["Maker of tea."]
    }]
  }]
}

